### PR TITLE
4 adding jsp and controllers

### PR DESCRIPTION
**Pull request by AnaLanda**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/cart/AddProductToCartController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductToCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/cart/AddProductToCartController.java:20]: I was told that it's better not to make service objects final.
Let's change this

File: src/main/java/com/internet/shop/controllers/cart/DeleteProductFromCartController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteProductFromCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/cart/DeleteProductFromCartController.java]: Same

File: src/main/java/com/internet/shop/controllers/InjectDataController.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+public class InjectDataController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/InjectDataController.java:24]: Same

File: src/main/java/com/internet/shop/controllers/cart/DisplayCartController.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DisplayCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/cart/DisplayCartController.java:17]: Same

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/product/AddProductController.java:15]: Same

File: src/main/java/com/internet/shop/controllers/product/GetAllProductsController.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllProductsController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/product/GetAllProductsController.java:16]: Same

File: src/main/java/com/internet/shop/controllers/user/DeleteUsersController.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteUsersController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/user/DeleteUsersController.java:17]: Same

File: src/main/java/com/internet/shop/controllers/user/GetAllUsersController.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllUsersController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/user/GetAllUsersController.java:16]: Same

File: src/main/java/com/internet/shop/controllers/user/RegistrationController.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RegistrationController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/user/RegistrationController.java:19]: Same

File: src/main/java/com/internet/shop/controllers/user/RegistrationController.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String login = req.getParameter("login");
+        String password = req.getParameter("pwd");
+        String repeatPassword = req.getParameter("pwd-repeat");
+        if (password.equals(repeatPassword)) {
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/user/RegistrationController.java:34]: What happens if you receive the empty Strings?

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/products/add.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/products/add.jsp").forward(req, resp);
+        String name = req.getParameter("name");
+        String productPrice = req.getParameter("price");
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/controllers/product/AddProductController.java:28]: What if those Strings are empty?

File: src/main/webapp/WEB-INF/views/cart/products.jsp [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    </div>
+    <br />
+    <br />
+    <a href="${pageContext.request.contextPath}/products/all">Add more items to your cart!</a>
+    <br />
+    <br />
+    <a href="${pageContext.request.contextPath}/">Return to the main page</a>
+</div>
+</body>
+</html>
```
Comment (dariiabutorina) [inline src/main/webapp/WEB-INF/views/cart/products.jsp]: Blank line

File: src/main/webapp/WEB-INF/views/products/add.jsp [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    <h1>Please enter the details of the product you'd like to add to our store.</h1>
+    <form method="post" action="${pageContext.request.contextPath}/products/add">
+        name: <input type="text" required="required" name="name">
+        price: <input type="number" required="required" name="price">
+        <button type="submit">add</button>
+    </form>
+    <a href="${pageContext.request.contextPath}/">Return to the main page</a>
+</div>
+</body>
+</html>
```
Comment (dariiabutorina) [inline src/main/webapp/WEB-INF/views/products/add.jsp]: Blank line

**Review by dariiabutorina — CHANGES_REQUESTED** [review]
Not bad overall, let's fix some little issues :)

File: src/main/java/com/internet/shop/controllers/user/RegistrationController.java (hunk 10 above)
Comment (AnaLanda) [reply src/main/java/com/internet/shop/controllers/user/RegistrationController.java:34]: Jsp form tag "required" will make this impossible)

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java (hunk 11 above)
Comment (AnaLanda) [reply src/main/java/com/internet/shop/controllers/product/AddProductController.java:28]: Jsp form tag "required" will make this impossible)

File: src/main/java/com/internet/shop/controllers/cart/AddProductToCartController.java (hunk 1 above)
Comment (AnaLanda) [reply src/main/java/com/internet/shop/controllers/cart/AddProductToCartController.java:20]: It is final already, I think. Do you mean it should be constant?)

Comment (dariiabutorina) [reply src/main/java/com/internet/shop/controllers/cart/AddProductToCartController.java:20]:
```suggestion
    private ShoppingCartService shoppingCartService =
            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
    private ProductService productService =
            (ProductService) injector.getInstance(ProductService.class);
```

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java (hunk 11 above)
Comment (mikhail771) [reply src/main/java/com/internet/shop/controllers/product/AddProductController.java:28]: Если придираться, то юзер может нажать правой кнопкой на поле для ввода имени, потом "посмотреть код" и там вручную удалить required и спокойно отправить на бэк пустое поле. Не знаю, как в рамках этой задачи, но вообще, на бэке тоже надо все проверять

Comment (RomanDubovskyi) [reply src/main/java/com/internet/shop/controllers/product/AddProductController.java:28]: we will learn about validation on back end later on, now solution with jsp is fine

File: src/main/java/com/internet/shop/controllers/user/DeleteUsersController.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
+        Long id = Long.valueOf(userId);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/user/DeleteUsersController.java:23]: let's delete shoppingCart here as well

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -10,6 +10,87 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addProductToCart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.cart.AddProductToCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addProductToCart</servlet-name>
+        <url-pattern>/cart/buy</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/cart/products/add</url-pattern>
```

**Review by RomanDubovskyi — COMMENTED** [review]
let's add controller to remove product from shoppingCart

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**